### PR TITLE
Fix: use `isinstance` instead of type comparison

### PR DESCRIPTION
**Pull request by melopeo**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by lostella — DISMISSED** [review]

File: src/gluonts/core/serde/_base.py [hunk 1]
```diff
@@ -309,13 +309,13 @@ def decode(r: Any) -> Any:
...
     # structural recursion over the possible shapes of r
-    if type(r) == dict and "__kind__" in r:
+    if isinstance(r, dict) and "__kind__" in r:
         kind = r["__kind__"]
         cls = cast(Any, locate(r["class"]))
 
         assert cls is not None, f"Can not locate {r['class']}."
 
-        if kind == Kind.Type:
+        if kind.value == Kind.Type.value:
```
Comment (abdulfatir) [inline src/gluonts/core/serde/_base.py]: What happens if you reset only this line to what it was before?

File: src/gluonts/core/serde/_base.py [hunk 2]
```diff
@@ -309,13 +309,13 @@ def decode(r: Any) -> Any:
...
     # structural recursion over the possible shapes of r
-    if type(r) == dict and "__kind__" in r:
+    if isinstance(r, dict) and "__kind__" in r:
         kind = r["__kind__"]
         cls = cast(Any, locate(r["class"]))
 
         assert cls is not None, f"Can not locate {r['class']}."
 
-        if kind == Kind.Type:
+        if kind == Kind.Type.value:
```
Comment (lostella) [inline src/gluonts/core/serde/_base.py]: I think this is not needed

**Review by lostella — CHANGES_REQUESTED** [review]
See comment

File: src/gluonts/core/serde/_base.py (hunk 2 above)
Comment (melopeo) [reply src/gluonts/core/serde/_base.py]: If I understand correctly, this line as it was originally, i.e. `if kind == Kind.Type:` is not passing the style checks. Any suggestions on how to fix it?

Comment (lostella) [reply src/gluonts/core/serde/_base.py]: I think it's passing them, is just a comparison between strings as far as I understand

File: src/gluonts/core/serde/_base.py (hunk 1 above)
Comment (melopeo) [reply src/gluonts/core/serde/_base.py]: In the original state this line was not passing the style checks.

File: src/gluonts/core/serde/_base.py (hunk 2 above)
Comment (melopeo) [reply src/gluonts/core/serde/_base.py]: ok. I will reset it to the original case. Maybe I just got it wrong.

File: src/gluonts/core/serde/_base.py (hunk 1 above)
Comment (abdulfatir) [reply src/gluonts/core/serde/_base.py]: I think this line had no issues with the checks. It was only the `type() == ` lines.

**Review by abdulfatir — APPROVED** [review]

**Review by lostella — APPROVED** [review]

**Pull request merged**